project/MiMaFilters.scala

```diff
@@ -3,28 +3,18 @@ import com.typesafe.tools.mima.core._
 
 object MiMaFilters {
   val Library: Seq[ProblemFilter] = Seq(
+    ProblemFilters.exclude[Problem]("scala.runtime.*"), // KEEP: scala.runtime isn't public API
```

Comment:
I would list these ones explicitly to make it clear to the reviewers which new stable methods were added.

Reply:
I disagree.  Can't reviewers just see the rest of the diff for what was added?  I want to make this a list of exceptional breakages to the binary backwards/both APIs, with minimal interaction.

Reply:
> Can't reviewers just see the rest of the diff for what was added? 

Yes, but it would be simpler for the reviewer to miss the fact that they should consider binary compatibility.

> I want to make this a list of exceptional breakages to the binary backwards/both APIs, with minimal interaction.

I'm worried about someone removing a method from the runtime once we do not generate code that calls that method. This would easily pass our tests and pass unnoticed. But if someone uses the current library to run code compiled against an older version of the compiler it would fail.

I would expect this list to never be large. We should never remove methods and new methods should be added as experimental. We would only stabilize the method in `scala.runtime` when passing to a new minor release, which would be tested under `backwards` and not need to additional fiilter.

Reply:
Right now it does not filter anything.

Reply:
> > Can't reviewers just see the rest of the diff for what was added?
> 
> Yes, but it would be simpler for the reviewer to miss the fact that they should consider binary compatibility.
> 
> > I want to make this a list of exceptional breakages to the binary backwards/both APIs, with minimal interaction.
> 
> I'm worried about someone removing a method from the runtime once we do not generate code that calls that method. This would easily pass our tests and pass unnoticed. But if someone uses the current library to run code compiled against an older version of the compiler it would fail.

Hmm, that's a fair concern.  I guess we could look to making this an method/class adding filter instead, but I can't think on the top of my head what that is, so I'll just drop it.

> I would expect this list to never be large. We should never remove methods and new methods should be added as experimental. We would only stabilize the method in `scala.runtime` when passing to a new minor release, which would be tested under `backwards` and not need to additional fiilter.

This is an aside, but I don't think we need to experimental/stabilize scala.runtime seeing as it's a private API.

> Right now it does not filter anything.

Yeah, it was preemptive. 🙂